contents/blog/mobile-session-replay.md

```diff
@@ -0,0 +1,144 @@
+
+### 2. Performance
+
+Desktops are a lot more powerful than phones. Because of this, we need to be much more sensitive about performance. 
+
+If you've ever tried to record your phone screen, you would know its impact on performance. Apps take longer to load, animations become choppy, reactivity degrades, and your phone heats up.
+
+Any that degrades the experience this much is not an option for many developers.
+
+### 3. Privacy
+
+A big difference between web and mobile apps is the DOM. 
+
+In web apps, this provides a hierarchical tree structure that represents and uniquely identifies elements. The web also has standardized elements, like `<input type="password">`. Combining these makes it easier to identify, mask, and exclude sensitive elements on the web.
+
+Mobile doesn't have standardized structures or elements. Accessibility identifiers are also inconsistently implemented. This means identifying, masking, and excluding content is a lot more tricky.
+
+### 4. Testing
+
+We're big fans of [dogfooding](/product-engineers/dogfooding) at PostHog. Often, we are our own best users. As you might know, we don't have a mobile app. 
+
+During development, we would need to rely on demo and open source apps. We risk creating something that doesn't work well for larger, production-quality apps.
+
+Developing a high-quality mobile replay product means relying more on our users and their feedback.
+
+## The prerequisites for mobile session replay
+
+First, none of this would be possible without [Manoel](/community/profiles/30206). We had mobile experience, but not the dedicated mobile SDK experience needed for this big of a feature. Manoel had that experience and this was only possible thanks to him.
+
+When Manoel joined the first thing he worked on was [rewriting](https://github.com/PostHog/product-internal/issues/506) the SDKs in Kotlin and Swift, removing code we didn't use, improving tests, automating deployments, and making sure they worked with the latest platforms.
+
+We already had the other prerequisite, our existing session replay infrastructure. We had the data structure, the way to store replays, as well as a complete product for playback and analysis. All this could be reused for mobile replay.
+
+![Replay](https://res.cloudinary.com/dmukukwp6/image/upload/replay_03a8c56981.png)
+
+Importantly, Paul and Manoel realized the mobile data needed to be transformed into a format the rrweb player could use. They wanted to keep the rrweb schema to ensure the fewest changes possible to our API and player. To do this, they wrote a [validation and testing tool](https://github.com/PostHog/mobile-replay-data-transformer) to rapidly test the transformations before deploying it in our main app. 
+
+Once this was done, everything was ready for the mobile replay capture to be worked on. 
+
+## How we built mobile session replay
```

Comment:
I was confused to read the heading, since I thought we were alreadt talking about how we are building this

Maybe my suggestion below works better?
```suggestion
## The process of building it
```

Reply:
Tbh I kind of feel like this section breaks the rhythym of this blog. See comment below on formatting